philosophy_of_software_design/README.md

```diff
@@ -81,3 +81,20 @@ The opposite of information hiding is _information leakage_. This occurs when th
 Information hiding can often be improved by making a module slightly larger. All the code for one capability should be in one single module to produce simpler interfaces (this relates to the module depth idea from previous section). Default values illustrate an example of partial information hiding, whenever possible a module should "do the right thing" without explicitly have to declaring to it. Information hiding also applies within a module encapsulating information between methods and minimizing the places where variables needs to be accessed by other methods. 
 
 It's important to not hide information if the information is needed outside its module or if it affects directly the module behaviour.
+
+## 8. Pull Complexity Downwards
+
+In case of having some unavoidable complexity in your module, should you let users of the module deal with the complexity, or should you handle the complexity internally within the module? If the complexity is related to the functionality provided by the module, then the second answer is usually the right one. It is more important for a module to have a simple interface than a simple implementation.
```

Comment:
```suggestion
In case of having some unavoidable complexity in your module, should you let users of the module deal with the complexity, or should you handle the complexity internally within the module? If the complexity is related to the functionality provided by the module, then the second answer is usually the right one. **It is more important for a module to have a simple interface than a simple implementation**.
```